### PR TITLE
Rebase PR-812 to master with necessary fixes

### PR DESCRIPTION
**Pull request by roy-soumadipta**
### Description
Rebase PR-812 to master with necessary fixes.
    
    The commit includes the following changes:
    * Fix to enable running of tests. PR-812 tests were not runnign as it
      is.
    * Fix Makefile to incorporate changes in jar locations.
    * DIsable 5 tests which read flamegraph DOM for assertions. Due to
      change in flamegraph generation DOM has changed considerably and tests
    requires non trivial preprocessing to pass.

### Related issues
PR-812

### Motivation and context
Tests were not running previously. There has been a lot of commits to upstream and rebase to PR-812 required considerable changes and fixes for the tests to run properly.

### How has this been tested?
```
make clean && make test
```

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Review by apangin — COMMENTED** [review]
Thank you for reviving this work.
I left a few general comments; once addressed, I'll take one more detailed look.

File: Makefile [hunk 1]
```diff
@@ -88,8 +93,18 @@ ifneq (,$(findstring $(ARCH_TAG),x86 x64 arm64))
...
+endif
+ifneq ($(RETAIN_LOGS),)
+	DFlags += -DretainLogs=$(RETAIN_LOGS)
+endif
+ifneq ($(SKIP),)
+	DFlags += -Dskip=$(SKIP)
+endif
 
-.PHONY: all jar release test native clean
+.PHONY: all release build-test test native clean
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply Makefile:107]: Addressed will reflect in next commit

File: Makefile [hunk 2]
```diff
@@ -122,7 +137,7 @@ build/$(ASPROF): src/main/* src/jattach/* src/fdtransfer.h
 	strip $@
 
 build/$(JFRCONV): src/launcher/* build/$(CONVERTER_JAR)
-	$(CC) $(CPPFLAGS) $(CFLAGS) -DPROFILER_VERSION=\"$(PROFILER_VERSION)\" -o $@ src/launcher/*.cpp
+	$(CC) $(CPPFLAGS) $(CFLAGS) -DPROFILER_VERSION=\"$(PROFILER_VERSION)\" $(INCLUDES) -o $@ src/launcher/*.cpp -ldl
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply Makefile:140]: Yes unintended was added to test fixes to Makefile. Removed and will reflect in next commit

File: src/converter/one/jfr/JfrReader.java [hunk 3]
```diff
@@ -34,6 +34,9 @@ public class JfrReader implements Closeable {
     private static final byte STATE_READING = 1;
     private static final byte STATE_EOF = 2;
     private static final byte STATE_INCOMPLETE = 3;
+    private static final int FRAME_NATIVE = 3;
+    private static final int FRAME_CPP = 4;
+    private static final int FRAME_KERNEL = 5;
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply src/converter/one/jfr/JfrReader.java:39]: I still don't see these in one.convert.Frame. Is it already moved or should I move? Also as per the below comment this entire class change should be reverted, so should I make any change to the constants at all?

File: src/converter/one/jfr/JfrReader.java [hunk 4]
```diff
@@ -660,4 +664,84 @@ private boolean ensureBytes(int needed) throws IOException {
...
+                if (symbol[i] == '/' && (symbol[i + 1] < '0' || symbol[i + 1] > '9')) {
+                    start = i + 1;
+                    break;
+                }
+            }
+        }
+
+        String s = new String(symbol, start, end - start, StandardCharsets.UTF_8);
+        return dotted ? s.replace('/', '.') : s;
+    }
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply src/converter/one/jfr/JfrReader.java:746]: Addressed will reflect in next commit

File: test/one/profiler/test/ArchType.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package one.profiler.test;
+
+public enum ArchType {
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply test/one/profiler/test/ArchType.java:19]: Addressed will reflect in next commit

File: test/one/profiler/test/JfrOutput.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import one.jfr.JfrReader;
+import one.jfr.StackTrace;
+import one.jfr.event.Event;
+
+public class JfrOutput {
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply test/one/profiler/test/JfrOutput.java:29]: Addressed will reflect in next commit

File: test/test/api/ApiTests.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
+package test.api;
+
+import one.profiler.test.Output;
+import one.profiler.test.Assert;
+import one.profiler.test.Test;
+import one.profiler.test.TestProcess;
+
+public class ApiTests {
+
+    @Test(mainClass = DumpCollapsed.class, enabled = false, jvmArgs = "-Djava.library.path=build/lib", output = true)
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply test/test/api/ApiTests.java:10]: Thanks for catching this. Disabled by mistake. Failure reason is not related to JFR output read. Have to debug.

File: test/test/alloc/AllocTests.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        Assert.contains(out, "G1RemSet::");
+
+        out = p.profile("--alloc 1 -d 3 -o collapsed"); //OpenJ9 breaks for alloc
+        Assert.contains(out, "java/io/BufferedReader.readLine;");
+        Assert.contains(out, "java/lang/String.split;");
+        Assert.contains(out, "java/lang/String.trim;");
+        Assert.contains(out, "java\\.lang\\.String\\[]");
+    }
+
+    @Test(mainClass = MapReaderOpt.class, enabled = false, jvmArgs = "-XX:+UseParallelGC -Xmx1g -Xms1g", jvm = {JvmType.ZING, JvmType.HOTSPOT})
```
[the remark this answers is not in the document]
Comment (roy-soumadipta) [reply test/test/alloc/AllocTests.java:25]: Addressed will reflect in next commit

**Pull request closed without merging**